### PR TITLE
Add failure message documentation

### PR DESCRIPTION
**Pull request by dottorblaster**
Just adding some failure message documentation.

### CODE REVIEW

**Comment by dottorblaster** [comment]
Formatting the file with `prettier` reformatted some of the existing lines. Should I keep it this way or revert it?

File: guides/specification.md [hunk 1]
```diff
@@ -513,6 +516,32 @@ Considering the previous scenario what happens is that:
 >     expect_same: facts.installed_rpm_version
 > ```
 
+### failure_message
+
+An optional failure message can be declare for every expectation.
```
Comment (arbulu89) [inline guides/specification.md:521]: `declare` -> `declared`

File: guides/specification.md [hunk 2]
```diff
@@ -513,6 +516,32 @@ Considering the previous scenario what happens is that:
...
+
+In case of an `expect` one, the failure message can interpolate `facts` and `values` present in the evaluation scope to provide more meaningful insights:
```
Comment (arbulu89) [inline guides/specification.md:523]: I would put "of an `expect` expectation".
I know it looks redundant, but is reads better than with `one`.

Besides that, `evalution scope` looks a bit technical. What about? `can interpolate facts and values present in the check definitition`

File: guides/specification.md [hunk 3]
```diff
@@ -513,6 +516,32 @@ Considering the previous scenario what happens is that:
...
+
+```yaml
+expectations:
+  - name: awesome_expectation
+    expect: values.awesome_constant_value == facts.awesome_fact
+    failure_message: The expectation did not match ${values.awesome_constant_value}
+```
+
+The outcome of the interpolation will be available into `ExpectationEvaluation` inside the API response.
```
Comment (arbulu89) [inline guides/specification.md:532]: Not a big a fan of future term. I would rephrase to `... interpolation is available into`

File: guides/specification.md [hunk 4]
```diff
@@ -513,6 +516,32 @@ Considering the previous scenario what happens is that:
...
+In case of an `expect_same` one, the failure message has to be a plain string:
+
+```yaml
+expectations:
+  - name: awesome_expectation
+    expect_same: facts.awesome_fact
+    failure_message: Boom!
+```
+
+This plain string will be available into `ExpectationResult` inside the API response.
```
Comment (arbulu89) [inline guides/specification.md:543]: The same for this `will`

**Review by arbulu89 — APPROVED** [review]
Great @dottorblaster ,
One simple typo to fix.
Everything else is personal preference.
PD: I'm personally fine with the prettier output
PD2: Before I forget, and as I don't know where to put. We should update the `demo` faker code so it sends the failure messages if it doesn't now. Maybe just changing the factories would do it
